src/main/java/com/faforever/client/api/JsonApiMessageConverter.java

```diff
@@ -28,17 +28,19 @@ public JsonApiMessageConverter(ResourceConverter resourceConverter) {
   @Override
   protected boolean supports(Class<?> clazz) {
     return Collection.class.isAssignableFrom(clazz)
-        || ReflectionUtils.getTypeName(clazz) != null;
+        || ReflectionUtils.getTypeName(clazz) != null
+        || clazz.equals(JSONAPIDocument.class);
   }
 
   @Override
   @SneakyThrows
-  @SuppressWarnings("unchecked")
   protected Object readInternal(Class<?> clazz, HttpInputMessage inputMessage) {
     try (InputStream inputStream = inputMessage.getBody()) {
       JSONAPIDocument<?> document;
       if (Iterable.class.isAssignableFrom(clazz)) {
         document = resourceConverter.readDocumentCollection(inputStream, Object.class);
+      } else if (clazz.equals(JSONAPIDocument.class)) {
+        return resourceConverter.readDocumentCollection(inputStream, Object.class);
```

Comment:
did i not push this to develop... in that case you should merge develop in or rebase